test/integration/consul-container/test/README.md

```diff
@@ -0,0 +1,123 @@
+4. Upgrade consul cluster and restart the envoy sidecars (we restart envoy sidecar to ensure the upgraded consul binary can read the state from the previous version and generate the correct envoy configurations) 
+5. Verify connection / disconnection (e.g., deny Action)  
+
+**Note** that all consul agents and user workloads such as application services, mesh-gateway are running in docker containers.
+
+
+The tests focus on Consul upgrade lifecycle and emit logs to stdout which developers can silence. Please reach out to the CTIA Team for any questions or help getting started.
+
+## Getting Started
+### Prerequisites
+If you wish to run or add new test cases, the following are required:
+- install [Go](https://go.dev/) (the version should match that of our CI config's Go image).
+- install [`golangci-lint`](https://golangci-lint.run/usage/install/) 
+- install [`Makefile`](https://www.gnu.org/software/make/manual/make.html)
+- [`Docker`](https://docs.docker.com/get-docker/) required to run tests locally  
+
+### Running Upgrade integration tests
+In CI, the tests are executed against all test cases defined under test/integration/consul-container/test/upgrade. To run a selected test case locally, e.g., testing upgrade from 1.15 to local changed version, we can use the following command:
+```sh
+make dev-docker
+cd ./test/integration/consul-container
+go test -v -timeout 30m -run ^TestACL_Upgrade_Node_Token$ ./.../upgrade/ --target-image consul --target-version local --latest-image consul --latest-version 1.15 -follow-log=false
+```
+
+The command above runs a single test. To run the entire upgrade test suite, the following command can be used.
+```sh
+go test -v -timeout 30m ./.../upgrade --target-image consul --target-version local --latest-image consul --latest-version 1.15 -follow-log=false
+```
+Below are the supported CLI options
+| Tags / Flags      | Default value | Description |
+| -----------         | ----------- | ----------- |
+| --latest-image      | consul    | Refers to the docker image name to be deployed before upgrade
+| --latest-version    | latest      | Refers to the version or tag of latest image to be deployed before upgrade
+| --target-image      | consul    | Refers to the image name that will be upgrade to
+| --target-version    | local     | Refers to the version of target image to upgrade to
+| -follow-log         |  false    | This optional flag emit event logs
+
+
+
+## Adding a new upgrade integration test  
+Upgrade integration tests are defined in the [test/integration/consul-container/test/upgrade](/test/integration/consul-container/test/upgrade) subdirectory and new upgrade integration tests should always be added to this location. The test framework uses
+[functional table-driven tests in Go](https://yourbasic.org/golang/table-driven-unit-test/) and using function types to modify the base value for each test case.
+
+For tests with multiple test cases, it should always start by invoking 
```

Comment:
If we are going to standardize on this form (a question for another thread maybe), we should adapt the others to follow suit.